prow/spyglass/artifacts.go

```diff
@@ -50,29 +50,47 @@ func (s *Spyglass) ListArtifacts(ctx context.Context, src string) ([]string, err
 	}
 
 	artifactNames, err := s.StorageArtifactFetcher.artifacts(ctx, gcsKey)
-	logFound := false
-	for _, name := range artifactNames {
-		if name == "build-log.txt" {
-			logFound = true
-			break
-		}
+	if err != nil {
+		return artifactNames, fmt.Errorf("error retrieving artifact names from gcs storage: %v", err)
+	}
+
+	jobName, buildID, err := s.KeyToJob(src)
+	if err != nil {
+		return artifactNames, fmt.Errorf("error parsing src: %v", err)
 	}
-	if err != nil || !logFound {
-		artifactNames = append(artifactNames, "build-log.txt")
+
+	job, err := s.jobAgent.GetProwJob(jobName, buildID)
+	if err != nil {
+		return artifactNames, nil
+	}
+
+	jobContainers := job.Spec.PodSpec.Containers
+
+	for _, container := range jobContainers {
+		logName := SingleLogName
+		if len(jobContainers) > 1 {
+			logName = fmt.Sprintf("%s-build-log.txt", container.Name)
+		}
+		if !contains(artifactNames, logName) {
```

Comment:
Not sure if it makes sense here already, but generally there is `k8s.io/apimachinery/pkg/util/sets` which makes working with sets easier

Reply:
Yes I agree it would probably makes sense to use sets instead. I'll make that change.